src/it/scala/org/ergoplatform/it/HistoryConsistencySpec.scala

```diff
@@ -0,0 +1,51 @@
+package org.ergoplatform.it
+
+import java.io.File
+
+import com.typesafe.config.Config
+import org.ergoplatform.it.container.{IntegrationSuite, Node}
+import org.scalatest.{FreeSpec, OptionValues}
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class HistoryConsistencySpec
+  extends FreeSpec
+    with IntegrationSuite
+    with OptionValues {
+
+  val shutdownAtHeight: Int = 5
+
+  val localVolume = s"$localDataDir/history-consistency-spec/data"
+  val remoteVolume = "/app"
+
+  val dir = new File(localVolume)
+  dir.mkdirs()
+
+  val nodeConfig: Config = specialDataDirConfig(remoteVolume)
+    .withFallback(shortDelayConfig)
+    .withFallback(offlineGeneratingPeerConfig)
+    .withFallback(nodeSeedConfigs.head)
+
+  val node: Node = docker.startNode(nodeConfig, specialVolumeOpt = Some((localVolume, remoteVolume))).get
+
+  "History consistency after unexpected shutdown" in {
+
+    val result = node.waitForHeight(shutdownAtHeight, 100.millis)
+      .flatMap(_ => node.waitForInconsistentHistory)
+      .flatMap { _ =>
+        docker.forceStopNode(node.containerId)
+        val restartedNode = docker
+          .startNode(nodeConfig, specialVolumeOpt = Some((localVolume, remoteVolume))).get
+        restartedNode.historyInfo
+          .flatMap(hi => restartedNode.waitForHeight(hi.bestBlockHeight + 1, 100.millis))
+          .flatMap(_ => restartedNode.historyInfo)
+          .flatMap { hi =>
+            restartedNode.headerIdsByHeight(hi.bestBlockHeight)
+              .map(_.headOption.value shouldEqual hi.bestBlockId)
```

Comment:
As you wrote in https://github.com/ergoplatform/ergo/issues/525, it is now valid situation, that bestFullBlock differs from bestHeader. It is not clear, what ar eyou actually testing here?

Reply:
This code tests the case when the node is shut down unexpectedly at the moment header and all its sections are stored but history indexes aren't updated yet - here we make sure that after such occasion history data is consistent after node restart.

Reply:
@catena2w, Test, used for i525 reproduction is different from this one, it exploits extremely short mining delay to create different blocks at the same height.